lib/improver/tests/psychrometric_calculations/psychrometric_calculations/test_PhaseChangeLevel.py

```diff
@@ -430,87 +484,133 @@ def setUp(self):
+            name='wet_bulb_temperature_integral', units='K m',)
+        wet_bulb_integral = add_coordinate(
+            wet_bulb_integral, [0, 1], 'realization')
+        self.wet_bulb_integral_cube_inverted = add_coordinate(
+            wet_bulb_integral, height_points[0:2], 'height',
+            coord_units='m', attributes=height_attribute)
+        self.wet_bulb_integral_cube_inverted.data = wbti_data
+        self.wet_bulb_integral_cube = sort_coord_in_cube(
+            self.wet_bulb_integral_cube_inverted, 'height', order='descending')
 
     def test_basic(self):
         """Test that process returns a cube with the right name and units."""
         self.orog.data[1, 1] = 100.0
-        result = FallingSnowLevel().process(
-            self.temperature_cube, self.relative_humidity_cube,
-            self.pressure_cube, self.orog, self.land_sea)
+        result = PhaseChangeLevel(phase_change='snow-sleet').process(
+            self.wet_bulb_temperature_cube, self.wet_bulb_integral_cube,
+            self.orog, self.land_sea)
         expected = np.ones((2, 3, 3), dtype=np.float32) * 66.88566
         self.assertIsInstance(result, iris.cube.Cube)
-        self.assertEqual(result.name(), "falling_snow_level_asl")
+        self.assertEqual(result.name(), "altitude_of_snow_falling_level")
         self.assertEqual(result.units, Unit('m'))
         self.assertArrayAlmostEqual(result.data, expected)
 
+    def test_inverted_input_cube(self):
+        """Test that the phase change level process returns a cube
+        containing the expected data when the height coordinate is in
+        ascending order rather than the expected descending order."""
+        self.orog.data[1, 1] = 100.0
+        result = PhaseChangeLevel(phase_change='snow-sleet').process(
+            self.wet_bulb_temperature_cube,
+            self.wet_bulb_integral_cube_inverted,
+            self.orog, self.land_sea)
+        expected = np.ones((2, 3, 3), dtype=np.float32) * 66.88566
+        self.assertArrayAlmostEqual(result.data, expected)
+
     def test_data(self):
-        """Test that the falling snow level process returns a cube
+        """Test that the phase change level process returns a cube
         containing the expected data when points at sea-level."""
         expected = np.ones((2, 3, 3), dtype=np.float32) * 65.88566
         orog = self.orog
         orog.data = orog.data * 0.0
         orog.data[1, 1] = 100.0
         land_sea = self.land_sea
         land_sea = land_sea * 0.0
-        result = FallingSnowLevel().process(
-            self.temperature_cube, self.relative_humidity_cube,
-            self.pressure_cube, orog, land_sea)
+        result = PhaseChangeLevel(phase_change='snow-sleet').process(
+            self.wet_bulb_temperature_cube, self.wet_bulb_integral_cube,
+            orog, land_sea)
         self.assertIsInstance(result, iris.cube.Cube)
         self.assertArrayAlmostEqual(result.data, expected)
 
```

Comment:
Please can you add one "sleet-rain" test too.

Reply:
Done